### PR TITLE
Swerve

### PR DESCRIPTION
**Pull request by JakeStew81**
# Pull request for Swerve

## Included Features:

- Swerve drivetrain
- Vision & odometry

## Checklist:

- [x] The code builds
- [x] The code has been self or peer-reveiwed
- [ ] The code has been tested on the robot & works as intended

### CODE REVIEW

**Comment by JakeStew81** [comment]
Wait until swerve has been tested

File: src/main/java/frc/lib/util/Wheel.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package frc.lib.util;
+
+import edu.wpi.first.units.Distance;
+import edu.wpi.first.units.Measure;
+
+public class Wheel {
+  public Measure<Distance> circumfrence;
```
Comment (eskiles) [inline src/main/java/frc/lib/util/Wheel.java]: Change these to public final to make the data object immutable.

File: src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -171,82 +159,76 @@ public static class ShooterConstants {
...
-            .currentLimit(60)
-            .idleMode(IdleMode.kBrake)
-            .build();
-    public static final MotorConfig TURN =
-        new MotorConfig.Builder()
-            .inverted(false)
-            .currentLimit(30)
-            .idleMode(IdleMode.kBrake)
-            .build();
+            new Translation2d[] {
```
Comment (eskiles) [inline src/main/java/frc/robot/Constants.java]: What is the ordering of these swerve pods?

**Review by eskiles — COMMENTED** [review]
Kinematics ordering.
Wheel public properties should be final.

**Pull request merged**